File: gaia-sdk-javascript/src/graphql/request/type/KnowledgeEdge.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, Timestamp, Struct, Long} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";

/**
 * Container type for static information
 */
export class KnowledgeEdge extends Array<(_:VariableRegistry) => string> {

    public source = () => { 
        this.push(_ => "source")
    };

    public target = () => { 
        this.push(_ => "target")
    };

    public type = () => { 
        this.push(_ => "type")
    };

    public weight = () => { 
        this.push(_ => "weight")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/KnowledgeEdge.py


from typing import Callable
from gaia_sdk.api.VariableRegistry import VariableRegistry


class KnowledgeEdge(list):
    """
    Container type for static information
    """

    def source(self):
        self.append(lambda x: "source")

    def target(self):
        self.append(lambda x: "target")

    def type(self):
        self.append(lambda x: "type")

    def weight(self):
        self.append(lambda x: "weight")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
